alica_engine/include/engine/PlanRepository.h

```diff
@@ -42,59 +43,100 @@ class PlanRepository {
+
+            iterator& operator++() {
+                ++_innerIter;
+                return *this;
+            }
+
+        private:
+            typename MapType<T>::const_iterator _innerIter;
+        };
+
+    public:
+        Accessor(const MapType<T>& map)
+                : _ref(map) {}
+
+        const T* operator[](int64_t id) const { return find(id); }
+        const T* find(int64_t id) const {
+            typename MapType<T>::const_iterator it = _ref.find(id);
+            return (it == _ref.end() ? nullptr : it->second);
+        }
+        Accessor(const Accessor&) = delete;
+        Accessor(Accessor&&) = default;
+
+        Accessor& operator=(const Accessor&) = delete;
+        Accessor& operator=(Accessor&&) = delete;
+
+        iterator begin() const { return iterator(_ref.begin()); }
+        iterator end() const { return iterator(_ref.end()); }
+
+    private:
+        const MapType<T>& _ref;
+    };
+
+    const Accessor<BehaviourConfiguration> getBehaviourConfigurations() const {
+        return Accessor<BehaviourConfiguration>(_behaviourConfigurations);
+    }
+    const Accessor<Behaviour> getBehaviours() const { return Accessor<Behaviour>(_behaviours); }
+    const Accessor<Capability> getCapabilities() const { return Accessor<Capability>(_capabilities); }
+    const Accessor<Characteristic> getCharacteristics() const { return Accessor<Characteristic>(_characteristics); }
+    const Accessor<EntryPoint> getEntryPoints() const { return Accessor<EntryPoint>(_entryPoints); }
+    const Accessor<Plan> getPlans() const { return Accessor<Plan>(_plans); }
+    const Accessor<PlanType> getPlanTypes() const { return Accessor<PlanType>(_planTypes); }
+    const Accessor<Quantifier> getQuantifiers() const { return Accessor<Quantifier>(_quantifiers); }
+    const Accessor<RoleDefinitionSet> getRoleDefinitionSets() const {
+        return Accessor<RoleDefinitionSet>(_roleDefinitionSets);
+    }
+    const Accessor<Role> getRoles() const { return Accessor<Role>(_roles); }
+    const Accessor<State> getStates() const { return Accessor<State>(_states); }
+    const Accessor<SyncTransition> getSyncTransitions() const { return Accessor<SyncTransition>(_syncTransitions); }
+    const Accessor<TaskRepository> getTaskRepositorys() const { return Accessor<TaskRepository>(_taskRepositories); }
+    const Accessor<Task> getTasks() const { return Accessor<Task>(_tasks); }
+    const Accessor<Transition> getTransitions() const { return Accessor<Transition>(_transitions); }
+    const Accessor<Variable> getVariables() const { return Accessor<Variable>(_variables); }
+    const Accessor<PlanningProblem> getPlanningProblems() const { return Accessor<PlanningProblem>(_planningProblems); }
+
+    PlanRepository(const PlanRepository&) = delete;
+    PlanRepository(PlanRepository&&) = delete;
+
+    PlanRepository& operator=(const PlanRepository&) = delete;
+    PlanRepository& operator=(PlanRepository&&) = delete;
```

Comment:
Doesn't seem like explicit delete of move operations is needed https://stackoverflow.com/questions/33096653/make-a-class-non-copyable-and-non-movable

Reply:
Maybe. But I do not trust compilers to be this correct wrt to the standard.
I know the microsoft compiler doesn't (or didn't) quite uphold that list.